pkg/nsx/services/vpc/builder.go

```diff
@@ -69,10 +70,10 @@ func buildNSXVPC(obj *v1alpha1.NetworkInfo, nsObj *v1.Namespace, nc v1alpha1.VPC
 			vpc.LoadBalancerVpcEndpoint = &model.LoadBalancerVPCEndpoint{Enabled: &loadBalancerVPCEndpointEnabled}
 		}
 	} else {
-		// for creating vpc case, fill in vpc properties based on networkconfig
-		vpcName := util.GenerateIDByObjectByLimit(obj, common.MaxSubnetNameLength)
-		vpc.DisplayName = &vpcName
-		vpc.Id = common.String(util.GenerateIDByObject(obj))
+		// for creating vpc case, fill in vpc properties based on networkconfig. Note, the NetworkInfo and the Namespace
+		// have the same value on the Name field, here we use the Namespace object to generate the NSX Vpc id and display_name.
+		vpc.DisplayName = common.String(s.buildVpcName(nsObj))
```

Comment:
Here, we use nsObj object, do you think we should crate a new obj with named as NetworkInfo but, UID is the same as namespace UID?

Reply:
I think the NetworkInfo CR's name is the same as the Namespace, no? https://github.com/vmware-tanzu/nsx-operator/blob/main/pkg/controllers/namespace/namespace_controller.go#L73